### ISSUE TITLE
Bundled common stylesheets/fonts for multiple SVGs

### ISSUE THREAD

**Issue by hsfzxjy** [terrastruct/d2#1187]
Hi, I am using D2 to generate multiple SVG diagrams that would be placed on the same HTML page. I observed the generated SVGs would share some common styles/font data, which seems redundant and will bloat the size of the HTML file. Is it possible to extract the common stylesheets/font data for multiple SVGs?

**Comment by alixander** [comment]
unfortunately not. the styles/fonts are quite minimal and diagram-specific. so for example, if you only have the test "asdf" appearing in your diagram, only the glyphs for "asdf" will be embedded. the styles are also theme-specific, and while in theory can be shared across diagrams which have the same themes, the savings there would be minimal. 

we have more aggressive optimizations on the sizing on the way though!

**Comment by hsfzxjy** [comment]
Alright. But apart from single-diagram optimization, I think it's also deserved to optimize targeting multiple diagrams in the web scenario. It's not rare for the diagrams on a web page to share the same theme/fonts, which could be extracted to for much more saving especially when the number of svg is large. 

Perhaps I could develop a separate tool as a prototype for this (it's a rather urged demand of mine).

**Comment by alixander** [comment]
@hsfzxjy can I ask what the urgent use case is? feel free to email me alex@terrastruct.com

**Comment by hsfzxjy** [comment]
I was occasionally writing some technical articles that are rendered as single html file which would embed many diagrams as illustration. Previously I used graghviz but it was really quirky. 

Some time ago I decided to switch to d2, but observed it generates much larger result than graphviz. The size of styles/fonts might be small for a single diagram, but it becomes innegligible when the number of svg goes up. I would like to keep the final html file as small as possible, especially those room could have been saved.

**Comment by alixander** [comment]
Gotcha. What version are you on? 0.3.0 saw a major decrease in size, just want to make sure you're at least on that

**Comment by hsfzxjy** [comment]
It's the latest v0.4.0.

Also I observe most floating point numbers in the generated svg are of high precision such as 4 or 6 decimal places due to the %f format spec, which might not be necessary. Changing this should be easy and  could lead to substantial size reduction for complicated paths like those generated by "sketch" mode.

**Comment by alixander** [comment]
yeah that's certainly one of the mentioned optimizations we plan on. going under the umbrella of "minimizing svg" (https://github.com/terrastruct/d2/issues/6).

our priorities are other requests that paying customers are asking for. i'm happy to review any PRs if you'd like to make any, and help point/guide you. otherwise, we'll probably have time to get to it sometime in the next month. 

(i know @donglixiaoche is also looking at more aggressive font subsetting which should reduce another 1/3 on the fonts)
